### PR TITLE
Add OneToOne example

### PR DESCRIPTION
**Pull request by jefffischer**
BroadleafCommerce/MicroPM#3607

### CODE REVIEW

**Comment by jefffischer** [comment]
@nathandmoore Is the example provided here a fair facsimile of your use case reported in the ticket? The reason I ask is because I never really saw the exception reported there. I did, however, see that the mapping was not applied. The fix seemed to be relatively simple in DataTracking.

Note, I also brought this example up in the admin and put it through a sandbox workflow lifecycle.

**Comment by nathandmoore** [comment]
Yes, this looks like what I tried when I got the detached entity exception. If you modify the Product again, do you not get any errors? What about when creating an application-level override?

**Comment by nathandmoore** [comment]
I was seeing an error when a new sandbox record was created for the Product when a production record already existed after fixing the mapping problem. That's I switched to having the Characteristic use the same id as the owning Product.

**Comment by jefffischer** [comment]
Thanks for the additional detail. I'll try these other cases as well and report back here.

**Comment by jefffischer** [comment]
No matter what I try - I don't seem to be able to break it with the change to DT in place. Here are the scenarios I tested:

Tenant level
- Existing production product with no Characteristic record at all. Specify characteristic fields in the admin and save, deploy.
- Edit, save, promote, deploy twice
- Create a new product with Characteristics. Save, promote, deploy. Edit again, promote, deploy

Application level
- Do all the steps from above
- Edit an inherited production record with no characteristic association. Save, promote, deploy.
- Edit an inherited production record with a characteristic defined.

**Comment by nathandmoore** [comment]
I'll retry it tomorrow or Friday. Maybe I can spot some difference

**Comment by jefffischer** [comment]
In the github issue, you'll see the three PRs that are relevant when you go to give it another try.

**Comment by nathandmoore** [comment]
Created a PR here with my changes https://github.com/BroadleafCommerce/MicroservicesDemo/pull/254 recreating the issue. The `postConvert` is now being used, but the detached error occurs on subsequent updates to a product.

**Comment by nathandmoore** [comment]
I'm not really sure where to look next for how to resolve this or why it's a different result than in the example of this PR. @jefffischer Worth talking about this when you get free from Data Feeds and other urgent activities.

**Comment by jefffischer** [comment]
Waiting to merge this for an official release of DT. At that point, I'll update the DT dependency in this PR to the GA release.

File: concepts/00400-productExtensionComplexFieldTableBased/src/main/java/com/tutorial/domain/Characteristics.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import static com.broadleafcommerce.common.jpa.JpaConstants.CONTEXT_ID_LENGTH;
+
+@Entity
+@Table(name = "CHARACTERISTICS")
+@Inheritance(strategy = InheritanceType.JOINED)
+@ToString
+@EqualsAndHashCode(exclude = {"id", "car"})
+public class Characteristics {
+
+    @Id
```
Comment (nathandmoore) [inline concepts/00400-productExtensionComplexFieldTableBased/src/main/java/com/tutorial/domain/Characteristics.java:36]:
```suggestion
    /**
     * Notice that we are not using {@code @GeneratedValue} but using the product's id as the
     * primary key. This is necessary for managing the subordinate entity in a {@code @OneToOne}
     * relationship in order for sandboxing transitions to function appropriately.
     */
    @Id
```

File: concepts/00400-productExtensionComplexFieldTableBased/src/main/java/com/tutorial/domain/Characteristics.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    @Column(name = "ELECTRIC_CAR_ID", nullable = false, length = CONTEXT_ID_LENGTH)
+    private String id;
+
+    @OneToOne(cascade = CascadeType.REFRESH)
+    @JoinColumn(name = "ELECTRIC_CAR_ID")
+    @MapsId
```
Comment (nathandmoore) [inline concepts/00400-productExtensionComplexFieldTableBased/src/main/java/com/tutorial/domain/Characteristics.java]:
```suggestion
    @MapsId // maps the product's id as the characteristic's primary key as well
```

**Review by nathandmoore — APPROVED** [review]
Want to reinforce what's happening with code comments

**Pull request merged**